src/libraries/System.Text.Json/tests/Common/RequiredKeywordTests.cs

```diff
@@ -143,6 +143,26 @@ public async Task ClassWithRequiredKeywordAndSmallParametrizedCtorFailsDeseriali
             Assert.Contains("Info2", exception.Message);
         }
 
+        [Fact]
+        public async Task InheritedPersonWithRequiredMembersWorksAsExpected()
+        {
+            var options = new JsonSerializerOptions(Serializer.DefaultOptions) { IncludeFields = true };
+            options.MakeReadOnly();
+
+            JsonTypeInfo typeInfo = options.GetTypeInfo(typeof(InheritedPersonWithRequiredMembers));
+            Assert.Equal(3, typeInfo.Properties.Count);
+
+            AssertJsonTypeInfoHasRequiredProperties(GetTypeInfo<InheritedPersonWithRequiredMembers>(options),
+                nameof(InheritedPersonWithRequiredMembers.FirstName),
+                nameof(InheritedPersonWithRequiredMembers.LastName));
+
+            await Assert.ThrowsAsync<JsonException>(() => Serializer.DeserializeWrapper("{}", typeInfo));
+        }
+
+        public class InheritedPersonWithRequiredMembers : PersonWithRequiredMembers
```

Comment:
Could you also add a test where 

1. a derived class is defining its own required members?
2. a dervived class using a constructor with `SetsRequiredMemberAttribute` set.?

Reply:
@eiriktsarpalis, tests was added by https://github.com/dotnet/runtime/pull/89418/commits/276c4fb35039c5ab8250b9295594c647a0f859cf .
Some existed tests of STJ had little bug with float number conversion. I had fixed this bug by https://github.com/dotnet/runtime/pull/89418/commits/7068976415da53b5a90438599f9f0b71ca17a752